### ISSUE TITLE
Can we get rid of the dependency on typecheck-decorator?

### ISSUE THREAD

**Issue by funkyfuture** [un-def/polypie#1]
ciao, polypie does exactly what i'm looking for. unfortunately it doesn't with Python3.6:

```
Traceback (most recent call last):
  File "/home/user/dev/polypie/test_polypie.py", line 101, in test_typing_annotations
    self.assertEqual(f(120, [1, 2, 3]), 'Any, Sequence')
  File "/home/user/dev/polypie/polypie.py", line 62, in wrapper
    return _call_func(func_key, args, kwargs)
  File "/home/user/dev/polypie/polypie.py", line 44, in _call_func
    return typecheck(func)(*args, **kwargs)
  File "/home/user/dev/polypie/.tox/py36/lib/python3.6/site-packages/typecheck/decorators.py", line 26, in typecheck
    checker = fw.Checker.create(v)
  File "/home/user/dev/polypie/.tox/py36/lib/python3.6/site-packages/typecheck/framework.py", line 154, in create
    if predicate(annotation):
  File "/home/user/dev/polypie/.tox/py36/lib/python3.6/site-packages/typecheck/typing_predicates.py", line 155, in _is_tg_union
    return issubclass(annotation, tg.Union)
  File "/usr/lib/python3.6/typing.py", line 799, in __subclasscheck__
    raise TypeError("Unions cannot be used with issubclass().")
TypeError: Unions cannot be used with issubclass().
```

looking at the repo-activity and the code of `typecheck-decorator`, i'd say it'd be better if polypie implemented  the needed functionality itself.

i'd dig into this, but a short hint what polypie essentially expects from calling `typecheck` in `_call_func` would be very helpful.

**Comment by un-def** [comment]
Hi! Thanks for reporting the issue.

`typecheck` is only used to check `typing` annotations (e.g., `def f(a: typing.Tuple[int, str])`). This is a lazy and expensive solution. I'll see later what I can do.

**Comment by un-def** [comment]
It took a long time, but we finally got rid of `typecheck-decorator`… sort of. I've just released version 0.2.0 where `typecheck-decorator` has been replaced with `typeguard`. Yeah, just another external dependency, but at least it works fine with the latest Python versions (3.5-3.7).

In case you're still interested in the project, I've opened another issue to remove any external typechecker completely: https://github.com/un-def/polypie/issues/3
